### PR TITLE
rustdoc: deprecate `#![doc(passes, plugins, no_default_passes)]`

### PR DESCRIPTION
**Pull request by QuietMisdreavus**
Closes https://github.com/rust-lang/rust/issues/48164

Blocked on https://github.com/rust-lang/rust/pull/50541 - this includes those changes, which were necessary to create the UI test

cc https://github.com/rust-lang/rust/issues/44136

Turns out, there were special attributes to mess with rustdoc passes and plugins! Who knew! Since we deprecated the CLI flags for this functionality, it makes sense that we do the same for the attributes.

This PR also introduces a `#![doc(document_private_items)]` attribute, to match the `--document-private-items` flag introduced in https://github.com/rust-lang/rust/pull/44138 when the passes/plugins flags were deprecated.

I haven't done a search to see whether these attributes are being used at all, but if the flags were any indication, i don't expect to see any users of these.

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
Resolved the merge conflict (but it still needs the other PR to land first).

**Comment by GuillaumeGomez** [comment]
Still waiting for the other PR to get merged first as well. :)

File: src/test/rustdoc-ui/deprecated-attrs.stderr [hunk 1]
```diff
@@ -0,0 +1,9 @@
+warning: WARNING: the `#![doc(no_default_passes)]` attribute is considered deprecated
```
[the remark this answers is not in the document]
Comment (QuietMisdreavus) [reply src/test/rustdoc-ui/deprecated-attrs.stderr]: Whoops, missed that one, haha.

**Comment by kennytm** [comment]
#50541 has been merged.

**Pull request merged**